lib/note-content-editor.jsx

```diff
@@ -171,6 +220,29 @@ export default class NoteContentEditor extends React.Component {
 		)
 	}
 
+	handleReturn = () => {
+		// matches lines that start with `- ` or `* `
+		// preceded by 0 or more tab characters
+		const listItemRe = /^\t*[-*]\s/;
```

Comment:
Is a line a valid list if it starts with a dash or star but no space? What about spaces instead of tabs? I thought spaces were valid too